diff --git a/kernel/Kconfig b/kernel/Kconfig
index 1234567..abcdefg 100644
--- a/kernel/Kconfig
+++ b/kernel/Kconfig
@@ -38,4 +38,87 @@ config KSU_X86_PATCH_SYSCALL_DISPATCHER
 	default n
 	help
 	  Dynamically patch x64's hardened syscall dispatcher to support syscall hooks. This is a
 	  replacement for a kernel source code patch, and is useful for x86_64 LKM mode.
+
+menu "KernelSU - SUSFS"
+config KSU_SUSFS
+	bool "KernelSU addon - SUSFS"
+	depends on KSU
+	default y
+	help
+	  Patch and Enable SUSFS to kernel with KernelSU
+
+config KSU_SUSFS_SUS_PATH
+	bool "Enable to hide suspicious path"
+	depends on KSU_SUSFS
+	default y
+	help
+	  Allow hiding the existence of user-defined file/directory from various system calls
+
+config KSU_SUSFS_SUS_MOUNT
+	bool "Enable to hide suspicious mounts"
+	depends on KSU_SUSFS
+	default y
+	help
+	  Allow hiding the user-defined mount paths from /proc/self/[mounts|mountinfo|mountstat]
+
+config KSU_SUSFS_SUS_KSTAT
+	bool "Enable to spoof suspicious kstat"
+	depends on KSU_SUSFS
+	default y
+	help
+	  Allow spoofing the kstat of user-defined file/directory
+
+config KSU_SUSFS_SUS_MAPS
+	bool "Enable to spoof suspicious maps"
+	depends on KSU_SUSFS
+	default y
+	help
+	  Allow spoofing the user-defined mappings in /proc/self/[maps|smaps|map_files]
+
+config KSU_SUSFS_SUS_PROC_FD_LINK
+	bool "Enable to spoof suspicous symbolic links (experimental)"
+	depends on KSU_SUSFS
+	default n
+	help
+	  Allow spoofing the user-defined symbolic links in /proc/self/fd/ and /proc/self/maps
+
+config KSU_SUSFS_SUS_MEMFD
+	bool "Enable to intercept memfd_create"
+	depends on KSU_SUSFS
+	default n
+	help
+	  Intercept memfd_create to allow hiding/create memory-backed files
+
+config KSU_SUSFS_TRY_UMOUNT
+	bool "Enable to auto-umount whenKSU process exits"
+	depends on KSU_SUSFS
+	default n
+	help
+	  Auto-umount when KSU process exits
+
+config KSU_SUSFS_SPOOF_UNAME
+	bool "Enable to spoof uname"
+	depends on KSU_SUSFS
+	default y
+	help
+	  Spoof uname to avoid detection
+
+config KSU_SUSFS_ENABLE_LOG
+	bool "Enable to print debug log"
+	depends on KSU_SUSFS
+	default n
+	help
+	  Enable SUSFS debug log
+
+config KSU_SUSFS_HIDE_KSU_SUSFS_SYMBOLS
+	bool "Enable to hide KSU/SUSFS symbols"
+	depends on KSU_SUSFS
+	default y
+	help
+	  Hide KSU and SUSFS symbols from /proc/kallsymbols
+
+config KSU_SUSFS_SPOOF_CMDLINE_OR_BOOTCONFIG
+	bool "Enable to spoof cmdline or bootconfig"
+	depends on KSU_SUSFS
+	default n
+	help
+	  Spoof /proc/cmdline or bootconfig to avoid detection
+
+config KSU_SUSFS_OPEN_REDIRECT
+	bool "Enable to redirect access to sensitive files"
+	depends on KSU_SUSFS
+	default y
+	help
+	  Redirect access to sensitive files like /system/build.prop
+
+endmenu
