--- a/selinux/rules.c
+++ b/selinux/rules.c
@@ -161,6 +161,9 @@
     ksu_destroy_sepolicy(old_pol);
 
     reset_avc_cache();
+#ifdef CONFIG_KSU_SUSFS
+    susfs_set_batch_sid();
+#endif
 out_unlock:
     mutex_unlock(&selinux_state.policy_mutex);
 }
@@ -520,6 +523,9 @@
     ksu_destroy_sepolicy(old_pol);
 
     reset_avc_cache();
+#ifdef CONFIG_KSU_SUSFS
+    susfs_set_batch_sid();
+#endif
     ret = success_cmd_count;
     goto out_unlock;
 
